client/components/command-palette/test/index.tsx

```diff
@@ -0,0 +1,142 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { act, fireEvent, render, screen } from '@testing-library/react';
+import React from 'react';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
+import { getCurrentRoutePattern } from '../../../state/selectors/get-current-route-pattern';
+import CommandPalette from '../index';
+import { useCommandPalette } from '../use-command-palette';
+
+const INITIAL_STATE = {
+	sites: {
+		items: {},
+	},
+	currentUser: {
+		capabilities: {
+			[ 1 ]: {
+				manage_options: true,
+			},
+		},
+	},
+	ui: {
+		selectedSiteId: 1,
+	},
+};
+
+const commands = [
+	{
+		name: 'getHelp',
+		label: 'Get help',
+		callback: ( { close }: { close: () => void } ) => {
+			close();
+		},
+	},
+	{
+		name: 'sendFeedback',
+		label: 'Send feedback',
+		searchLabel: 'find help',
+		callback: ( { close }: { close: () => void } ) => {
+			close();
+		},
+	},
+	{
+		name: 'clearCache',
+		label: 'Clear cache',
+		callback: ( { close }: { close: () => void } ) => {
+			close();
+		},
+	},
+	{
+		name: 'enableEdgeCache',
+		label: 'Enable edge cache',
+		callback: ( { close }: { close: () => void } ) => {
+			close();
+		},
+	},
+];
+const mockStore = configureStore();
+const store = mockStore( INITIAL_STATE );
+const queryClient = new QueryClient();
+
+jest.mock( '../../../state/selectors/get-current-route-pattern' );
+jest.mock( '../use-command-palette' );
+
+window.ResizeObserver = jest.fn( () => ( {
+	observe: jest.fn(),
+	unobserve: jest.fn(),
+	disconnect: jest.fn(),
+} ) );
+
+describe( 'CommandPalette', () => {
+	( getCurrentRoutePattern as jest.Mock ).mockReturnValue( '/sites' );
+
+	const renderCommandPalette = () => {
+		( useCommandPalette as jest.Mock ).mockReturnValue( {
+			commands: commands,
+			filterNotice: 'Mock Filter Notice',
+			emptyListNotice: 'No results found',
+		} );
+
+		render(
+			<Provider store={ store }>
+				<QueryClientProvider client={ queryClient }>
+					<CommandPalette />
+				</QueryClientProvider>
+			</Provider>
+		);
+
+		act( () => {
+			fireEvent.keyDown( document, { key: 'k', metaKey: true } );
+		} );
+	};
+
+	it( 'should confirm that the command palette opens with the commands from the commands array', () => {
+		renderCommandPalette();
+
+		expect( screen.getByPlaceholderText( 'Search for commands' ) ).toBeInTheDocument();
+		expect( screen.getByText( 'Get help' ) ).toBeInTheDocument();
+	} );
+
+	it( 'should return only "Get help" command as it matches label and "Send feedback" as it matches searchLabel; other commands are hidden', () => {
+		renderCommandPalette();
+
+		expect( screen.getByPlaceholderText( 'Search for commands' ) ).toBeInTheDocument();
+		fireEvent.change( screen.getByPlaceholderText( 'Search for commands' ), {
+			target: { value: 'help' },
+		} );
+
+		expect( screen.getByText( 'Get' ) ).toBeInTheDocument();
```

Comment:
This looks strange but this is because when you add something to the search field, it shows up bold so it adds markdown around a label that it matches so the output ends up being something like 'Get <mark>help<mark>' so I am targeting the two words separately

Reply:
Another alternative could be targeting the element from document.

```
expect( document.getElementById( 'get-help' ) ).toBeInTheDocument();
```

or

```
expect( document.querySelector( '[data-value="get help"]' ) ).toBeInTheDocument();
```

No need to modify it or create a follow-up PR.